### PR TITLE
[e2e] External - Expand WPCOM suite, part 4

### PR DESCRIPTION
**Pull request by Stojdza**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Updates to the current `e2e-pw` setup, so tests are passing against the WPCOM environment. 

In this PR, we're covering `/merchant` folder to work with WPCOM. 

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. CI green - to be sure that changes in this PR are not affecting wp-env execution.
2. Locally - checkout this branch and run `pnpm test:e2e:with-env default-wpcom`. In `playwright.config.js` you can comment everything except this (to save some time for local execution):
```
'**/merchant/order-coupon.spec.js',
'**/merchant/order-edit.spec.js',
'**/merchant/order-emails.spec.js',
'**/merchant/order-refund.spec.js',
'**/merchant/order-search.spec.js',
'**/merchant/order-status-filter.spec.js',
'**/merchant/page-loads.spec.js',
'**/merchant/product-create-simple.spec.js',
'**/merchant/product-delete.spec.js',
'**/merchant/product-edit.spec.js',
```
3. In the code, I put comments on why these changes.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Expand the e2e suite we're running on WPCOM part #4. 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/order-coupon.spec.js [hunk 1]
```diff
@@ -119,40 +119,50 @@ test.describe(
...
-				page.getByRole( 'cell', {
-					name: `$${ discountedPrice }`,
-					exact: true,
-				} )
-			).toBeVisible();
-			// remove the coupon
-			await page.locator( 'a.remove-coupon' ).dispatchEvent( 'click' ); // have to use dispatchEvent because nothing visible to click on
+		test(
+			'can remove a coupon',
+			{ tag: [ '@skip-on-default-wpcom' ] },
```
Comment (Stojdza) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/order-coupon.spec.js:124]: Error 500

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/order-emails.spec.js [hunk 2]
```diff
@@ -113,7 +112,7 @@ test.describe(
 					.click();
 
 				await expect(
-					page.getByText( 'Receiver wordpress@example.com' )
+					page.getByText( `Receiver ${admin.email}` )
```
Comment (Stojdza) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/order-emails.spec.js]: Use `admin.email` so test is passing on WPCOm and Pressable now.

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/page-loads.spec.js [hunk 3]
```diff
@@ -201,29 +201,31 @@ for ( const currentPage of wcPages ) {
...
-					page,
-				} ) => {
-					await page
-						.locator(
-							`li.wp-menu-open > ul.wp-submenu > li a:text-is("${ currentPage.subpages[ i ].name }")`
-						)
-						.click();
+				test(
+					`Can load ${ currentPage.subpages[ i ].name }`,
+					{ tag: '@skip-on-default-wpcom' },
```
Comment (Stojdza) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/page-loads.spec.js:206]: Skipping this one. I will address it in a separate PR

**Comment by adimoldovan** [comment]
@Stojdza I cannot get these 2 tests to pass locally:

```
  2 failed
    [default wpcom] › merchant/order-emails.spec.js:122:3 › Merchant > Order Action emails received › can receive completed email
    [default wpcom] › merchant/product-edit.spec.js:192:1 › can restore regular price when bulk editing products
```

I ran the a few times, they keep failing.

**Comment by Stojdza** [comment]
> I cannot get these 2 tests to pass locally

@adimoldovan, it looks like these two will benefit from [sites cleanup](https://github.com/woocommerce/woocommerce/pull/51720) :) After I cleaned up the WPCOM site, tests passed multiple times in a row. I did clean up once again now. Can you, please, try one more time?

**Comment by adimoldovan** [comment]
> > I cannot get these 2 tests to pass locally
> 
> @adimoldovan, it looks like these two will benefit from [sites cleanup](https://github.com/woocommerce/woocommerce/pull/51720) :) After I cleaned up the WPCOM site, tests passed multiple times in a row. I did clean up once again now. Can you, please, try one more time?

Tried again, but one is still failing. There is an SMTP Error displayed because the email cannot be delivered (as expected). I'm not sure why this error is obscuring the expected `We have finished processing your order.` string from the email content, because in the screenshots I can see the string is there. I think the used locator needs an update for this case.

```
    [default wpcom] › merchant/order-emails.spec.js:122:3 › Merchant > Order Action emails received › can receive completed email


Error: Timed out 20000ms waiting for expect(locator).toContainText(expected)


Locator: locator('#wp-mail-logging-modal-content-body-content')
- Expected string  -  1
+ Received string  + 56

- We have finished processing your order.

+                         Time                    
+             2024-10-01 10:06:44        
+                         Receiver                    
+             john.doe.merchant.test@example.com        
+                         Subject                    
+             Your WooCommerce Core E2E Test Suite order is now complete        
+                         Error                    
+             SMTP Error: The following recipients failed: john.doe.merchant.test@example.com: : Recipient address rejected: Mailbox undeliverable. Expires: 2024-10-25 10:11:55
+                         This email failed to send. Install WP Mail SMTP to solve your deliverability issues.                    
+                     Dismiss this notice.
+                         Headers                    
+             Content-Type: text/html
+ Reply-to: woocommerce 
+                         Message      


            "              
```

**Comment by Stojdza** [comment]
It was happening for me that this string was different than expected, `Receiver john.doe.merchant.test@example.com`, but it is interesting that now it is passing for me but not for you. 

I'll take a look.

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/page-loads.spec.js [hunk 4]
```diff
@@ -87,8 +87,8 @@ const wcPages = [
 			{
 				name: 'Overview',
 				heading: 'Overview',
-				element: '.woocommerce-marketing-card-header-description',
-				text: 'Start by adding a channel to your store',
+				element: '.woocommerce-marketing-channels-card',
+				text: 'Channels',
```
Comment (Stojdza) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/page-loads.spec.js:91]: Based on the discussion here https://github.com/woocommerce/woocommerce/pull/51744#issuecomment-2385540520

**Review by adimoldovan — APPROVED** [review]
Looks good, all tests passed for me.

**Comment by Stojdza** [comment]
- Based on the discussion [here](https://github.com/woocommerce/woocommerce/pull/51744#issuecomment-2385540520), I've updated `page-loads.spec.js` in [46f675a](https://github.com/woocommerce/woocommerce/pull/51424/commits/46f675abfef07fa591ab1425b1c888438e71cf0a). This affects wp-env as well, but we should be good there. 

- Also related to the discussion [here](https://github.com/woocommerce/woocommerce/pull/51744#issuecomment-2385540520), @lanej0 and I concluded how to delete all coupons (including trashed ones), so the test(s) is passing if there are any _leftovers_. 

- Skipped ["can receive completed email"](https://github.com/woocommerce/woocommerce/pull/51424/commits/4765fe721c4f144f2da3ad283dd7ed458c3e37c2) for now. I couldn't determine why it started failing on WPCOM, so I'll address it separately.

**Pull request merged**